confmap/confmap.go

```diff
@@ -41,6 +41,10 @@ func NewFromStringMap(data map[string]any) *Conf {
 // The confmap.Conf can be unmarshalled into the Collector's config using the "service" package.
 type Conf struct {
 	k *koanf.Koanf
+	// If true, upon unmarshaling do not call the Unmarshal function on the struct
+	// if it implements Unmarshaler and is the top-level struct.
+	// This avoids running into a circular dependency.
```

Comment:
We should define what the circular dependency is here. I think calling out the potential for an infinite recursion caused by `Unmarshaler.Unmarshal` and `Conf.Unmarshal` calling each other would probably be a good mention.

Reply:
thanks, addressed.